Add property to disable table collapsing

Suvra19 opened the pull request:
### Problem
The collapsed view of the table works well for most use cases. However, some users might want to get a view of all the columns at once to analyse the table records. We encountered such a use case and thus this solution.

### Solution
Provide a `isCollapsible` flag/property that the client can use to opt-out of the collapsed view if required. This property is `true` be default. This flags enables the client to provide the user with an option to expand all the columns and analyse the table records much more conveniently.

### UI implications
Expanding the columns will lead to horizontal scrolling.

Salinn commented:
Hey @Suvra19 thanks for this PR! Sorry it has taken me a few days to get around to looking at this. I do see when I try running this locally an error in storybook. Any chance you could fix the Storybook example so that works for anyone viewing this and bump the version in the package.json?
![Uploading Screen Shot 2020-04-02 at 10.21.35 AM.png…]()

Salinn commented:
<img width="1679" alt="Screen Shot 2020-04-02 at 10 21 35 AM" src="https://user-images.githubusercontent.com/2657757/78261056-b9f1bf80-74cc-11ea-8d7a-76a755b553f3.png">

Suvra19 commented:
> Hey @Suvra19 thanks for this PR! Sorry it has taken me a few days to get around to looking at this. I do see when I try running this locally an error in storybook. Any chance you could fix the Storybook example so that works for anyone viewing this and bump the version in the package.json?

Hey @Salinn, ahh sorry I didn't notice that. I will have a look and fix it asap.

Suvra19 commented:
Hey @Salinn, I fixed the storybook error and also added one for demonstrate expanding all columns when `isCollapsible` flag is passed as false.  Also bumped the version. 

Please let me know if you have any questions.
